niap-cc/NIAPSecExample/app/src/main/java/com/android/certifications/niap/tests/SDPDeviceCredentialTestWorker.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.certifications.niap.tests;
+
+import android.content.Context;
+
+import com.android.certifications.niap.MainActivity;
+import com.android.certifications.niap.TestUtil;
+import com.android.certifications.niap.niapsec.SecureConfig;
+import com.android.certifications.niap.niapsec.biometric.BiometricSupport;
+import com.android.certifications.niap.niapsec.biometric.BiometricSupportImpl;
+import com.android.certifications.niap.niapsec.context.SecureContextCompat;
+import com.android.certifications.niap.niapsec.crypto.SecureKeyGenerator;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPairGenerator;
+import java.util.concurrent.Executors;
+
+import javax.crypto.Cipher;
+
+import androidx.annotation.NonNull;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+/**
+ * Worker class that runs a test to encrypt and decrypt data with the NIAPSEC library.
+ * <p>
+ * This test should be successful after running the application. Please note that, this test
+ * does require that the user authorize using the devices default unlock implementation which can
+ * be a device pin or password, fingerprint, or face identification.
+ */
+public class SDPDeviceCredentialTestWorker extends Worker {
+
+    private static final String FILE_NAME = "test_file";
+    private static final String KEY_PAIR_ALIAS = "key_pair_alias";
+
+    public SDPDeviceCredentialTestWorker(Context context, WorkerParameters parameters) {
+        super(context, parameters);
+
+        try {
+            new File(getApplicationContext().getFilesDir(), FILE_NAME).delete();
+        } catch (Exception ex) {
+        }
+    }
+
+    @NonNull
+    @Override
+    public Result doWork() {
+        try {
+            // Write SDP File
+            BiometricSupport biometricSupport = new BiometricSupportImpl(
+                    MainActivity.thisActivity,
+                    getApplicationContext(), true) {
+                @Override
+                public void onAuthenticationSucceeded() {
+                    TestUtil.logSuccess(getClass(), "SDP Device Credential Unlock " +
+                            "Succeeded, private key available for decryption through the " +
```

Comment:
Is this indentation right? Makes it seems like an argument even though it is a continuation of argument 2... what happens on AS when you hit enter?  

If that default, then ignore this comment.